std/algorithm/searching.d

```diff
@@ -2855,9 +2857,19 @@ if (isForwardRange!R1 && isForwardRange!R2)
             asTuple = rhs;
         }
         Tuple!(S1, S1, S2) asTuple;
-        bool opCast(T : bool)()
+        static if (hasConstEmptyMember!(typeof(asTuple[1])))
```

Comment:
I recall @WalterBright mentioning its on his list of todos.

Reply:
I'm afraid I don't know what is meant here. Is there a bugzilla issue for it?

Reply:
A quick search turns up

https://issues.dlang.org/show_bug.cgi?id=8407
https://issues.dlang.org/show_bug.cgi?id=7521
https://issues.dlang.org/show_bug.cgi?id=7521

I think that the idea would be that if every function the templated function called was `const`, then the templated function would be inferred as `const`, but I don't know what the ultimate consequences of that would be. There's no question that there are cases where that would be really useful, but given how restrictive `const` is, I have a nagging feeling that there's a risk that doing it would end up marking something as `const` when it shouldn't. At the moment, I don't see how though.

Reply:
Apart from checking for calls const-only functions you also have to check for absence of mutations via assignments aswell, right?

Reply:
Yeah, though technically, it could be argued that assignment would just mean calling `opAssign` which wouldn't be `const` and thus would already be violating the rule of calling a non-`const` function (though in the case of primitive types, I don't think that there's actually an `opAssign` in the current implementation, so such a rule might be necessary).